clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -157,14 +197,26 @@ object UncommittedGarbageCollector {
       firstSlice: String,
       startTime: java.time.Instant,
       success: Boolean,
+      expiredAddresses: DataFrame,
       removed: DataFrame
   ): Unit = {
-    val reportDst = s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
+    val reportDst = reportPath(storageNamespace, runID)
     writeJsonSummary(reportDst, runID, firstSlice, startTime, success, removed.count())
 
+    expiredAddresses.write.parquet(s"$reportDst/marked")
     removed.write.parquet(s"$reportDst/deleted")
   }
 
+  private def reportPath(storageNamespace: String, runID: String): String = {
+    s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
+  }
+
+  private def readMarkedAddresses(storageNamespace: String, markID: String): DataFrame = {
+    // TODO(eden) - check if this run succeeded
```

Comment:
Why is this a TODO? We already have everything we need to check it

Reply:
done